tensorflow/python/framework/sparse_tensor_test.py

```diff
@@ -97,6 +97,14 @@ def testConsumers(self):
       self.assertIn(dense.op, sp.consumers())
       self.assertIn(out.op, sp.consumers())
 
+  def testWithValues(self):
+      source = sparse_tensor.SparseTensor(
+          indices=[[0, 0], [1, 2]], values=[1., 2], dense_shape=[3, 4])
+      new_tensor = source.with_values([5.0, 1.0])
+      self.assertAllEqual(new_tensor.indices, source.indices)
+      self.assertAllEqual(new_tensor.values, [5.0, 1.0])
+      self.assertAllEqual(new_tensor.dense_shape, source.dense_shape)
+
```

Comment:
Please add a test that this fails if the values has inappropriate shape.  Something like:

```
self.assertRaises((errors.InvalidError, ValueError)):
  source.with_values([5.0, 1.0, 8.0])
```

(Where the two exceptions are because we get different errors depending on whether we're in graph mode or eager mode.)

Reply:
I've added the test in slightly modified form from what you've written, such that it not only looks at the number of values, but ensures equality of the shape as well.